Lib/test/test_fileio.py

```diff
@@ -73,6 +73,24 @@ def testBlksize(self):
             blksize = getattr(fst, 'st_blksize', blksize)
         self.assertEqual(self.f._blksize, blksize)
 
+    def testReadWithWritingMode(self):
+        r, w = os.pipe()
+        w = os.fdopen(w, "w")
+        w.write("hello")
+        w.close()
+        with io.FileIO(r, mode="w") as f:
+            with self.assertRaises(_io.UnsupportedOperation):
+                f.read()
+
+    def testReadallWithWritingMode(self):
```

Comment:
These new test methods have too much duplicated code.

Reply:
I (or @evanWang1409) can pull out the pipe setup into a separate method, if you would like. Otherwise I'm not sure what else is the right direction here. It's not clear what abstraction would make sense given that there are only two callers right now and this is a small test case.